.circleci/config.yml

```diff
@@ -0,0 +1,515 @@
+            JAR_BASE_NAME='<<parameters.jar_base_name>>'
+            JAR_BUILD_BUCKET_DIR='<<parameters.jar_build_bucket_dir>>'
+            JAR_FILE_NAME="${JAR_BASE_NAME}_${DATE}_${SHORT_GIT_SHA}.jar"
+            LOCAL_JAR_FILE_PATH="./target/${JAR_BASE_NAME}.jar"
+            JAR_BUCKET_URL="gs://${BUILDS_BUCKET}${JAR_BUILD_BUCKET_DIR}/${JAR_FILE_NAME}"
+            gsutil cp "${LOCAL_JAR_FILE_PATH}" "${JAR_BUCKET_URL}"
+            echo "${JAR_FILE_NAME} successfully uploaded to ${JAR_BUCKET_URL}"
+  
+  retrieve-jar:
+    description: Get <<parameters.jar_base_name>> jar from bucket
+    parameters:
+      jar_base_name:
+        type: string
+      builds_bucket:
+        type: string
+        default: *builds_bucket
+      jar_build_bucket_dir:
+        type: string
+        default: *jar_build_bucket_dir
+    steps:
+      - auth-gcp-service-account:
+          env: prod
+      - run:
+          name: Retrieve <<parameters.jar_base_name>> jar from bucket
+          command: |
+            set -u
+            set +e
+            BUILDS_BUCKET='<<parameters.builds_bucket>>'
+            JAR_BASE_NAME='<<parameters.jar_base_name>>'
+            JAR_BUILD_BUCKET_DIR='<<parameters.jar_build_bucket_dir>>'
+            JAR_FILE_URL_PATTERN="gs://${BUILDS_BUCKET}${JAR_BUILD_BUCKET_DIR}/${JAR_BASE_NAME}_*_${SHORT_GIT_SHA}.jar"
+            echo "Checking for ${JAR_FILE_URL_PATTERN}"
+            # For some reason following line generating exit code 1 every time in CircleCI. Use set +e so script can continue
+            JAR_FILE_URL=`gsutil ls $JAR_FILE_URL_PATTERN  | sort -r | head -1`
+            if [ -z $JAR_FILE_URL ]
+              then
+                echo "Could not find archive using pattern ${JAR_FILE_URL_PATTERN}"
+                exit 1
+            fi
+            JAR_FILE_PATH="target/${JAR_BASE_NAME}.jar"
+            mkdir -p $(dirname $JAR_FILE_PATH)
+            gsutil cp  $JAR_FILE_URL $JAR_FILE_PATH
+            echo ""
+  deploy-jar:
+    parameters:
+      jar_base_name:
+        type: string
+      deploy_dir:
+        type: string
+        default: deployment-assets
+    steps:
+      - set-deployment-env
+      - render-configs
+      - run:
+          name: Gather deployment assets
+          command: |
+            DEPLOY_DIR='<<parameters.deploy_dir>>'
+            rm -rf $DEPLOY_DIR
+            mkdir -p $DEPLOY_DIR
+            cd $DEPLOY_DIR
+            cp ../target/<<parameters.jar_base_name>>.jar .
+            cp ../src/appengine/<<parameters.jar_base_name>>.yaml .
+            cp ../output-config/application.conf .
+            cp -r ../output-config/fc_keys .
+      - auth-gcp-service-account
+      - run:
+          name: Deploy <<parameters.jar_base_name>> to GAE
+          command: |
+            set -u
+            gcloud app deploy --version="${SHORT_GIT_SHA}" --stop-previous-version --project "broad-ddp-${ENVIRONMENT}" <<parameters.deploy_dir>>/<<parameters.jar_base_name>>.yaml
```

Comment:
With jar in the target directory (there because we just built or because we pulled it out of our repo), we deployment to GAE.